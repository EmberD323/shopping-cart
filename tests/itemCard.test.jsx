
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import ItemCard from "../src/components/itemCard";


describe("Nav Bar component", () => {
    it("renders card", () => {
      // since screen does not have the container property, we'll destructure render to obtain a container for this test
      const { container } = render(<ItemCard />);
      expect(container).toMatchSnapshot();
    });
  
  //   it("renders Shop Page after link click", async () => {
  //     const user = userEvent.setup();
  
  //     render(<App />);
  //     const button = screen.getByRole("button", { name: "Click Me" });
  
  //     await user.click(button);
  
  //     expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  //   });
  });