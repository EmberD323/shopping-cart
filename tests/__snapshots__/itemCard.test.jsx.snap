// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav Bar component > renders card 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="itemTitle"
    />
    <div
      class="itemImage"
    />
    <div
      class="amount"
    >
      <button>
        -
      </button>
      <form>
        <label />
        <input />
      </form>
      <button>
        +
      </button>
    </div>
    <button>
      add to cart
    </button>
  </div>
</div>
`;

exports[`Nav Bar component > renders twolinks 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="itemTitle"
    />
    <div
      class="itemImage"
    />
    <div
      class="amount"
    >
      <button>
        -
      </button>
      <form>
        <label />
        <input />
      </form>
      <button>
        +
      </button>
    </div>
    <button>
      add to cart
    </button>
  </div>
</div>
`;
